src/nvim/api/win_config.c

```diff
@@ -187,22 +209,67 @@ Window nvim_open_win(Buffer buffer, Boolean enter, Dict(float_config) *config, E
   if (!parse_float_config(config, &fconfig, false, true, err)) {
     return 0;
   }
-  win_T *wp = win_new_float(NULL, false, fconfig, err);
+
+  bool is_split = HAS_KEY(config, float_config, split) || HAS_KEY(config, float_config, vertical);
+
+  win_T *wp = NULL;
+  if (is_split) {
+    win_T *parent = NULL;
+    if (!HAS_KEY(config, float_config, win) || config->win != -1) {
+      parent = find_window_by_handle(fconfig.window, err);
+      if (!parent) {
+        return 0;
+      } else if (parent->w_floating) {
+        api_set_error(err, kErrorTypeException, "Cannot split a floating window");
+        return 0;
+      }
+    }
+
+    if (HAS_KEY(config, float_config, vertical) && !HAS_KEY(config, float_config, split)) {
+      if (config->vertical) {
+        fconfig.split = p_spr ? kWinSplitRight : kWinSplitLeft;
+      } else {
+        fconfig.split = p_sb ? kWinSplitBelow : kWinSplitAbove;
+      }
+    }
+    int flags = win_split_flags(fconfig.split, parent == NULL);
+
+    if (parent == NULL) {
+      wp = win_split_ins(0, flags, NULL, 0);
+    } else {
+      tabpage_T *tp = win_find_tabpage(parent);
+      tabpage_T *old_curtab = curtab;
+      RedrawingDisabled++;
+      if (tp->handle != curtab->handle) {
+        goto_tabpage_tp(tp, false, false);
+      }
+      WIN_EXECUTE(parent, tp, {
+        wp = win_split_ins(0, flags, NULL, 0);
+      });
+      if (!enter && tp->handle != old_curtab->handle) {
+        goto_tabpage_tp(old_curtab, true, false);
+      }
+      RedrawingDisabled--;
```

Comment:
Thanks for the reviews! Yeah, this is definitely part of the code I wanted to revisit. This is the only way I could find that would allow me to insert a window in a specific location though - any ideas?

Especially for the case of the window being in another tabpage, I think that complicates things a good bit.